stdlib/public/core/IntegerParsing.swift

```diff
@@ -192,9 +81,315 @@ extension FixedWidthInteger {
   ///
   /// - Parameter description: The ASCII representation of a number.
   @inlinable
-  @_semantics("optimize.sil.specialize.generic.partial.never")
-  @inline(__always)
   public init?(_ description: String) {
     self.init(description, radix: 10)
   }
 }
+
+@inlinable @inline(__always)
+internal func _parseSmall<Result: FixedWidthInteger>(
+  _ rawGuts: _StringObject.RawBitPattern, radix: Int
+) -> Result? {
+  // We work with the bytes as they are arranged in the StringObject,
+  // i.e. the leading characters are in the least-significant bits.
+  // This is as opposed to SmallString order, which swaps them on big-endian
+  // platforms to maintain the same in-memory order.
+  var word1 = rawGuts.0
+  let word2 = rawGuts.1 & 0x00ff_ffff_ffff_ffff
+  let count = Int((rawGuts.1 &>> 56) & 0x0f)
```

Comment:
You can get a `_SmallString` struct from the guts, which has some of these operations on it. In general, we'd want to avoid duplicating these magic flag values.